gapis/resolve/framegraph.go

```diff
@@ -0,0 +1,45 @@
+// Copyright (C) 2020 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package resolve
+
+import (
+	"context"
+
+	"github.com/google/gapid/core/log"
+	"github.com/google/gapid/gapis/capture"
+	"github.com/google/gapid/gapis/database"
+	"github.com/google/gapid/gapis/service/path"
+)
+
+// Framegraph retrieves the framegraph from the database, creating it if need be.
+func Framegraph(ctx context.Context, p *path.Framegraph, r *path.ResolveConfig) (interface{}, error) {
+	obj, err := database.Build(ctx, &FramegraphResolvable{Capture: p.Capture})
```

Comment:
This is not necessary. Anything that is returned via the `Get` API is already stored in the database. Also, since you are calling `database.Build`, it's run in-place, so just put the code from the below `Resolve` here.

Reply:
Fixed with: https://github.com/google/agi/pull/542

Note that I was mirroring `gapis/resolve/resources.go`, which has this double-function layout .